### PR TITLE
[SYCL] Re-enable several joint matrix E2E tests

### PR DESCRIPTION
**Pull request by lbushi25**
Removed the XFAIL directive from several joint matrix tests as they are now passing on all devices except for DG2 GPU because apparently it does not yet support sub-group size of 32 for joint matrix and it also does not support VNNI transforms. Hence, these tests are marked as UNSUPPORTED on DG2 and on all other devices they are now expected to pass.

### CODE REVIEW

File: sycl/test-e2e/Matrix/SG32/joint_matrix_annotated_ptr.cpp [hunk 1]
```diff
@@ -5,15 +5,14 @@
...
+// VNNI transform and sub-group size 32 are not supported yet on DG2 by IGC
+// UNSUPPORTED: gpu-intel-dg2
 // REQUIRES: aspect-ext_intel_matrix
 // REQUIRES-INTEL-DRIVER: lin: 27501, win: 101.4943
 
 // RUN: %{build} -o %t.out
 // RUN: %{run} %t.out
 
-// Currently row major B fails when annotated_ptr is used
-// XFAIL: gpu
```
Comment (YuriPlyakhin) [inline sycl/test-e2e/Matrix/SG32/joint_matrix_annotated_ptr.cpp:15]: Does this test pass on PVC? I did not see any unexpected pass on PVC for this test...

File: sycl/test-e2e/Matrix/SG32/joint_matrix_bfloat16_colmajorA_colmajorB.cpp [hunk 2]
```diff
@@ -14,8 +16,6 @@
 // This tests support of col major layout for matrix B which does transpose and
 // then VNNI transform. This is currently only available on AMX
 
-// XFAIL: gpu
```
Comment (YuriPlyakhin) [inline sycl/test-e2e/Matrix/SG32/joint_matrix_bfloat16_colmajorA_colmajorB.cpp:17]: Does this test pass on PVC? I did not see any unexpected pass on PVC for this test...

File: sycl/test-e2e/Matrix/SG32/joint_matrix_colA_rowB_colC.cpp [hunk 3]
```diff
@@ -5,14 +5,14 @@
...
 //===----------------------------------------------------------------------===//
+// VNNI transform and sub-group size 32 are not supported yet on DG2 by IGC
+// UNSUPPORTED: gpu-intel-dg2
 // REQUIRES: aspect-ext_intel_matrix
 // REQUIRES-INTEL-DRIVER: lin: 27501, win: 101.4943
 
 // RUN: %{build} -o %t.out
 // RUN: %{run} %t.out
 
-// XFAIL:*
```
Comment (YuriPlyakhin) [inline sycl/test-e2e/Matrix/SG32/joint_matrix_colA_rowB_colC.cpp:14]: Does this test pass on PVC? I did not see any unexpected pass on PVC for this test...

File: sycl/test-e2e/Matrix/SG32/joint_matrix_int8_colmajorA_colmajorB.cpp [hunk 4]
```diff
@@ -14,8 +16,6 @@
 // This tests support of col major layout for matrix B which does transpose and
 // then VNNI transform. This is currently only available on AMX
 
-// XFAIL: gpu
```
Comment (YuriPlyakhin) [inline sycl/test-e2e/Matrix/SG32/joint_matrix_int8_colmajorA_colmajorB.cpp:17]: Does this test pass on PVC? I did not see any unexpected pass on PVC for this test...

File: sycl/test-e2e/Matrix/joint_matrix_bfloat16_colmajorA_colmajorB.cpp [hunk 5]
```diff
@@ -13,8 +15,6 @@
 // This tests support of col major layout for matrix B which does transpose and
 // then VNNI transform. This is currently only available on AMX
 
-// XFAIL: gpu
```
Comment (YuriPlyakhin) [inline sycl/test-e2e/Matrix/joint_matrix_bfloat16_colmajorA_colmajorB.cpp:16]: Does this test pass on PVC? I did not see any unexpected pass on PVC for this test...

File: sycl/test-e2e/Matrix/joint_matrix_bfloat16_colmajorA_colmajorB.cpp [hunk 6]
```diff
@@ -5,6 +5,8 @@
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 //
 //===----------------------------------------------------------------------===//
+// VNNI transform and sub-group size 32 are not supported yet on DG2 by IGC
```
Comment (YuriPlyakhin) [inline sycl/test-e2e/Matrix/joint_matrix_bfloat16_colmajorA_colmajorB.cpp:8]: This comment is wrong, as this test doesn't use sub-group size 32

File: sycl/test-e2e/Matrix/joint_matrix_colA_rowB_colC.cpp [hunk 7]
```diff
@@ -5,13 +5,13 @@
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 //
 //===----------------------------------------------------------------------===//
+// VNNI transform and sub-group size 32 are not supported yet on DG2 by IGC
```
Comment (YuriPlyakhin) [inline sycl/test-e2e/Matrix/joint_matrix_colA_rowB_colC.cpp:8]: This comment is wrong, as this test doesn't use sub-group size 32

File: sycl/test-e2e/Matrix/joint_matrix_colA_rowB_colC.cpp [hunk 8]
```diff
@@ -5,13 +5,13 @@
...
+// UNSUPPORTED: gpu-intel-dg2
 // REQUIRES: aspect-ext_intel_matrix
 
 // RUN: %{build} -o %t.out
 // RUN: %{run} %t.out
 
-// XFAIL:*
```
Comment (YuriPlyakhin) [inline sycl/test-e2e/Matrix/joint_matrix_colA_rowB_colC.cpp:13]: Does this test pass on PVC? I did not see any unexpected pass on PVC for this test...

File: sycl/test-e2e/Matrix/joint_matrix_int8_colmajorA_colmajorB.cpp [hunk 9]
```diff
@@ -5,6 +5,8 @@
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 //
 //===----------------------------------------------------------------------===//
+// VNNI transform and sub-group size 32 are not supported yet on DG2 by IGC
```
Comment (YuriPlyakhin) [inline sycl/test-e2e/Matrix/joint_matrix_int8_colmajorA_colmajorB.cpp:8]: This comment is wrong, as this test doesn't use sub-group size 32

File: sycl/test-e2e/Matrix/joint_matrix_int8_colmajorA_colmajorB.cpp [hunk 10]
```diff
@@ -13,8 +15,6 @@
 // This tests support of col major layout for matrix B which does transpose and
 // then VNNI transform. This is currently only available on AMX
 
-// XFAIL: gpu
```
Comment (YuriPlyakhin) [inline sycl/test-e2e/Matrix/joint_matrix_int8_colmajorA_colmajorB.cpp:16]: Does this test pass on PVC? I did not see any unexpected pass on PVC for this test...

**Review by YuriPlyakhin — CHANGES_REQUESTED** [review]
I think this change should not be committed. Please, check my comments. Thanks.

File: sycl/test-e2e/Matrix/SG32/joint_matrix_annotated_ptr.cpp [hunk 11]
```diff
@@ -5,15 +5,14 @@
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 //
 //===----------------------------------------------------------------------===//
+// VNNI transform and sub-group size 32 are not supported yet on DG2 by IGC
```
Comment (YuriPlyakhin) [inline sycl/test-e2e/Matrix/SG32/joint_matrix_annotated_ptr.cpp:8]: VNNI is not relevant for this test
```suggestion
// Sub-group size 32 is not supported yet on DG2 by IGC
```

File: sycl/test-e2e/Matrix/joint_matrix_bfloat16_colmajorA_colmajorB.cpp (hunk 6 above)
Comment (YuriPlyakhin) [inline sycl/test-e2e/Matrix/joint_matrix_bfloat16_colmajorA_colmajorB.cpp:8]: sub-group size 32 part is not relevant for this test.
```suggestion
// VNNI transform is not supported yet on DG2 by IGC
```

File: sycl/test-e2e/Matrix/SG32/joint_matrix_annotated_ptr.cpp (hunk 1 above)
Comment (lbushi25) [reply sycl/test-e2e/Matrix/SG32/joint_matrix_annotated_ptr.cpp:15]: I forgot to test this on PVC. I just tested it now and yes you are right, it fails. 
I'm adding an unsupported directive for pvc on these tests.

Comment (dkhaldi) [reply sycl/test-e2e/Matrix/SG32/joint_matrix_annotated_ptr.cpp:15]: The test is supported on PVC but it is XFAIL.
So you should keep both:
// UNSUPPORTED: gpu-intel-dg2
and
// XFAIL: gpu
(same goes to the other tests)

**Comment by lbushi25** [comment]
> I think this change should not be committed. Please, check my comments. Thanks.

Ok, closing this PR. Instead, I'm creating an issue to track the fail on PVC and later add a reference to that issue as a comment in the test.

**Pull request closed without merging**